### ISSUE TITLE
Error converting to onnx: forward function contains for loop

### ISSUE THREAD

**Issue by wytcsuch** [pytorch/pytorch#58289]
Hello, there is a for loop in my forward function. When I turned to onnx, the following error occurred:

`[ONNXRuntimeError] : 1 : FAIL : Non-zero status code returned while running Split node. Name:'Split_ 1277' Status Message: Cannot split using values in 'split' attribute. Axis=0 Input shape={59} NumOutputs=17 Num entries in 'split' (must equal number of outputs) was 17 Sum of sizes in 'split' (must equal size of selected axis) was 17`

Part of my forward code：
```
    y, ey, x, ex = pad(boxes, w, h)
    if len(boxes) > 0:
        im_data = []
        indx_y = torch.where(ey > y-1)[0]
        for ind in indx_y:
             img_k =  imgs[image_inds[ind],:, (y[ind] - 1).type(torch.int64):ey[ind].type(torch.int64), (x[ind]-1).type(torch.int64):ex[ind].type(torch.int64)].unsqueeze(0)
             im_data.append(imresample(img_k, (24, 24)))
        im_data = torch.cat(im_data, dim=0)
        return im_data
```
I found that during the first onnx conversion, the for loop was executed 17 times, but when I tested it, the for loop required 59 times, so there was an error. In the forward function, indx_y is dynamic, so the number of for loops is also dynamic. Is there any way to solve this problem？

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @SplitInfinity

**Comment by justinchuby** [comment]
We’ve gone ahead and closed this issue because it is **stale**.

If you still believe this issue is relevant, please feel free to reopen the issue and we will triage it as necessary. Please specify in a comment any updated information you may have so that we can address it effectively. We encourage you to try the latest pytorch-preview (nightly) version to see if it has resolved the issue.
